dbms/src/Storages/Transaction/RegionBlockReader.cpp

```diff
@@ -37,18 +38,51 @@ RegionBlockReader::RegionBlockReader(DecodingStorageSchemaSnapshotConstPtr schem
     : schema_snapshot{std::move(schema_snapshot_)}
 {}
 
+
 bool RegionBlockReader::read(Block & block, const RegionDataReadInfoList & data_list, bool force_decode)
 {
-    switch (schema_snapshot->pk_type)
+    try
+    {
+        switch (schema_snapshot->pk_type)
+        {
+        case TMTPKType::INT64:
+            return readImpl<TMTPKType::INT64>(block, data_list, force_decode);
+        case TMTPKType::UINT64:
+            return readImpl<TMTPKType::UINT64>(block, data_list, force_decode);
+        case TMTPKType::STRING:
+            return readImpl<TMTPKType::STRING>(block, data_list, force_decode);
+        default:
+            return readImpl<TMTPKType::UNSPECIFIED>(block, data_list, force_decode);
+        }
+    }
+    catch (DB::Exception & exception)
     {
-    case TMTPKType::INT64:
-        return readImpl<TMTPKType::INT64>(block, data_list, force_decode);
-    case TMTPKType::UINT64:
-        return readImpl<TMTPKType::UINT64>(block, data_list, force_decode);
-    case TMTPKType::STRING:
-        return readImpl<TMTPKType::STRING>(block, data_list, force_decode);
-    default:
-        return readImpl<TMTPKType::UNSPECIFIED>(block, data_list, force_decode);
+        // to print more info for debug the random ddl test issue(should serve stably when ddl #004#)
+        // https://github.com/pingcap/tiflash/issues/7024
+        auto print_column_defines = [&](const DM::ColumnDefinesPtr & column_defines) {
+            std::stringstream ss;
```

Comment:
better use `FmtBuffer` `dbms/src/Common/FmtUtils.h` for joining && formating the strings